bip-0141.mediawiki

```diff
@@ -83,7 +83,7 @@ If all transactions in a block do not have witness data, the commitment is optio
 
 === Witness program ===
 
-A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a 1-byte push opcode (for 0 to 16) followed by a data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
+A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a select subset of opcodes (<code>OP_0,OP_1,OP_2,...,OP_16</code>) followed by a data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
```

Comment:
> consists of a select subset of opcodes (<code>OP_0,OP_1,OP_2,...,OP_16</code>)

This is incorrect as it implies *any subset* of 17 opcodes. The specification is *any member* of that set. It's also not clear what is meant by "select subset".

> The value of the first push is called the "version byte".

It is unclear what is the value of `OP_0`, since it's push/stack value is `[]` (not `0`), which isn't a byte and therefore cannot be a "version byte".

I suggest:

> A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of one opcode from (<code>OP_0, OP_1, OP_2, ..., OP_16</code>) followed by a data push of between 2 and 40 bytes gets a new meaning. The logical value of the first push (i.e. [0..16]) is called the "script version". The stack value of the second push (of [2..40] bytes) is called the "witness program"

Reply:
I think that explanation is pretty good. A nit is `script version` -> `witness version`.